### PR TITLE
[3.2] Replace websocketpp with boost beast

### PR DESCRIPTION
**Pull request by ClaytonCalabrese**
> Overhauls http_plugin, replacing websocketpp with boost::beast. Feature parity with previous implementation, with exception of new additions keep-alive/multiple requests per session and additional server-side error logging. Additional test in plugin_http_api_test for keep-alive.
> https://github.com/EOSIO/eos/pull/10583

Resolves https://github.com/eosnetworkfoundation/mandel/issues/148

Previous discussions were had in the old repo:
https://github.com/eosnetworkfoundation/mandel/pull/675

Will merge https://github.com/AntelopeIO/fc/pull/4 and then update fc submodule.

### CODE REVIEW

**Comment by spoonincode** [comment]
Can you bump the fc submodule to the branch you have over in fc w/ websocketpp removed?

**Comment by spoonincode** [comment]
I've been eyeballing this some and I think we should go ahead and get it in. Can you please get it synced to main, make sure anything related to #64 is taken care of, and bump the fc submodule.

File: plugins/http_plugin/http_plugin.cpp [hunk 1]
```diff
@@ -762,47 +261,59 @@ class http_plugin_impl : public std::enable_shared_from_this<http_plugin_impl> {
...
+            my->plugin_state->valid_hosts.insert(aliases.begin(), aliases.end());
          }
 
+         my->plugin_state->thread_pool_size = options.at( "http-threads" ).as<uint16_t>();
+         EOS_ASSERT( my->plugin_state->thread_pool_size > 0, chain::plugin_config_exception,
+                     "http-threads ${num} must be greater than 0", ("num", my->plugin_state->thread_pool_size));
+
+         my->plugin_state->max_bytes_in_flight = options.at( "http-max-bytes-in-flight-mb" ).as<uint32_t>() * 1024 * 1024;
+         my->plugin_state->max_requests_in_flight = options.at( "http-max-in-flight-requests" ).as<int32_t>();
+         my->plugin_state->max_response_time = fc::microseconds( options.at("http-max-response-time-ms").as<uint32_t>() * 1000 );
```
Comment (heifner) [inline plugins/http_plugin/http_plugin.cpp:303]: All in this section above are duplicates.

**Comment by heifner** [comment]
@spoonincode If you are good we can merge https://github.com/AntelopeIO/fc/pull/4 update this PR and then merge it.

**Review by spoonincode — APPROVED** [review]
yep let's go ahead

**Pull request merged**